lightning/src/routing/router.rs

```diff
@@ -150,11 +160,124 @@ impl cmp::PartialOrd for RouteGraphNode {
 struct DummyDirectionalChannelInfo {
 	cltv_expiry_delta: u32,
 	htlc_minimum_msat: u64,
+	htlc_maximum_msat: Option<u64>,
 	fees: RoutingFees,
 }
 
+/// It's useful to keep track of the hops associated with the fees required to use them,
+/// so that we can choose cheaper paths (as per Dijkstra's algorithm).
+/// Fee values should be updated only in the context of the whole path, see update_value_and_recompute_fees.
+/// These fee values are useful to choose hops as we traverse the graph "payee-to-payer".
+#[derive(Clone)]
+struct PaymentHop {
+	route_hop: RouteHop,
+	/// Liquidity available considering the following limitations:
+	/// - UTXO capacity
+	/// - htlc_maximum_msat (direction-dependent)
+	/// - use of the same channel by other paths per MPP (including fees paid there)
+	/// It does not take into account fees to be paid on the current hop, so
+	/// this amount *should* cover transferring a value AND paying fees (on this channel).
+	available_liquidity_msat: u64,
+	/// Minimal fees required to route to the previous hop via any of its inbound channels.
+	src_lowest_inbound_fees: RoutingFees,
+	/// Fees of the channel used in this hop.
+	channel_fees: RoutingFees,
+	/// All the fees paid *after* this channel on the way to the destination
+	min_total_fee_to_payee: u64,
+	/// Fee paid for the use of the current channel (see channel_fees).
+	/// The value will be actually paid on the previous hop.
+	hop_use_fee_msat: u64,
+	/// Fee required to reach the source node of the current channel (estimate, see src_lowest_inbound_fees)
+	prev_hop_use_estimate_fee_msat: u64,
+}
+
+impl PaymentHop {
+	/// How attractive this channel is in terms of the paid fees.
+	fn get_fee_weight_msat(&self) -> u64 {
```

Comment:
Why total path? This is a fee at the hop, right? We use this to compare different hops considering how they are located in the path, in terms of fees etc.

Reply:
Oh, it seems to add together `fees paid up until this hop + fees at this hop + fees from this hop to the final destination` (which seems like the total path fees)? But I don't mind too much, feel free to leave it.

Reply:
I think the key misunderstanding is using `min_total_fee_to_payee` here. We may call it `estimated_min_total_path_fees_msat`

And even then, I think it's confusing because it's called on the hop, not on the path.